posts/2022-02-24-Rust-1.59.0.md

```diff
@@ -0,0 +1,242 @@
+---
+layout: post
+title: "Announcing Rust 1.59.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.59.0.
+Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust 1.59.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.59.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1590-2022-02-24
+
+## What's in 1.59.0 stable
+
+### Inline assembly
+
+The Rust language now supports inline assembly. This enables many applications
+of Rust that need very low-level control over their execution, or access to
+specialized machine instructions.
+
+When compiling for x86 and x86-64 targets, for instance, you can now write:
+
+```rust
+use std::arch::asm;
+
+// Multiply x by 6 using shifts and adds
+let mut x: u64 = 4;
+unsafe {
+    asm!(
+        "mov {tmp}, {x}",
+        "shl {tmp}, 1",
+        "shl {x}, 2",
+        "add {x}, {tmp}",
+        x = inout(reg) x,
+        tmp = out(reg) _,
+    );
+}
+assert_eq!(x, 4 * 6);
+```
+
+The format string syntax used to name registers in the `asm!` and `global_asm!`
+macros is the same used in Rust format strings, so it should feel quite familiar
+to Rust programmers.
+
+The syntax and instructions available in the assembly code itself vary according
```

Comment:
```suggestion
The assembly language and instructions available in the assembly code itself vary according
```

Maybe? Otherwise it might be interpreted as 'format string syntax' varies, which is not the case, I believe? Maybe we can also reword to something simpler, like "Each target has a fairly different set of instructions, so usage would often be cfg-gated in practice" or something like that?